src/EditorFeatures/CSharpTest/Diagnostics/Suppression/RemoveUnnecessaryPragmaSuppressionsTests.cs

```diff
@@ -0,0 +1,768 @@
+            protected abstract string VariableAssignedButNotUsedDiagnosticId { get; }
+            protected abstract ImmutableArray<string> UnsupportedDiagnosticIds { get; }
+
+            public sealed class CompilerTests : CompilerOrAnalyzerTests
+            {
+                internal override ImmutableArray<DiagnosticAnalyzer> OtherAnalyzers
+                    => ImmutableArray.Create<DiagnosticAnalyzer>(new CSharpCompilerDiagnosticAnalyzer());
+                protected override string VariableDeclaredButNotUsedDiagnosticId => "CS0168";
+                protected override string VariableAssignedButNotUsedDiagnosticId => "CS0219";
+                protected override ImmutableArray<string> UnsupportedDiagnosticIds
+                {
+                    get
+                    {
+                        var errorCodes = Enum.GetValues(typeof(ErrorCode));
+                        var supported = ((CSharpCompilerDiagnosticAnalyzer)OtherAnalyzers[0]).GetSupportedErrorCodes();
+                        using var _ = ArrayBuilder<string>.GetInstance(out var builder);
+                        foreach (int errorCode in errorCodes)
+                        {
+                            if (!supported.Contains(errorCode) && errorCode > 0)
+                            {
+                                // Add all 3 supported formats for suppressions: integer, integer with leading zeros, "CS" prefix
+                                builder.Add(errorCode.ToString());
+                                builder.Add(errorCode.ToString("D4"));
+                                builder.Add("CS" + errorCode.ToString("D4"));
+                            }
+                        }
+
+                        return builder.ToImmutable();
+                    }
+                }
+            }
+
+            public sealed class AnalyzerTests : CompilerOrAnalyzerTests
+            {
+                internal override ImmutableArray<DiagnosticAnalyzer> OtherAnalyzers
+                    => ImmutableArray.Create<DiagnosticAnalyzer>(new UserDiagnosticAnalyzer(), new CompilationEndDiagnosticAnalyzer());
+                protected override string VariableDeclaredButNotUsedDiagnosticId => UserDiagnosticAnalyzer.Descriptor0168.Id;
+                protected override string VariableAssignedButNotUsedDiagnosticId => UserDiagnosticAnalyzer.Descriptor0219.Id;
+                protected override ImmutableArray<string> UnsupportedDiagnosticIds
+                    => ImmutableArray.Create(
+                        CompilationEndDiagnosticAnalyzer.Descriptor.Id,
+                        IDEDiagnosticIds.RemoveUnnecessarySuppressionDiagnosticId,
+                        IDEDiagnosticIds.FormattingDiagnosticId,
+                        "format");
+            }
+
+            [Fact]
+            public async Task TestDoNotRemoveRequiredDiagnosticSuppression()
+            {
+                await TestMissingInRegularAndScriptAsync(
```

Comment:
not using the new test frameowkr.

Reply:
Yes, unfortunately because I need to pass in multiple analyzers here, which is not supported in the new testing library.

Reply:
got it.  thanks.